git-quick-stats

```diff
@@ -1004,6 +1027,20 @@ if [[ "$#" -eq 1 ]]; then
         -b|--branch-tree) branchTree;;
         -D|--branches-by-date) branchesByDate;;
         -C|--contributors) contributors;;
+        -n|--new-contributors)
+            newDate=""
+            while [[ -z "${newDate}" ]]; do
+                read -r -p "Since what date? " newDate
+                # Test if the date provide is valid and try again if it isn't.
+                # date(1) is pretty accepting of time stamps but you never know
+                # what people may try. This script doesn't provide any additional
+                # output for a bad date since `date`'s STDERR already contains
+                # useful information.
+                if [ ! $(date -d "${newDate}" +%s) ]; then
```

Comment:
I believe the more correct way of testing this would be:

```sh
if ! date -d "${newDate}" +%s > /dev/null 2>&1; then
    newDate=""
fi
```

Give this a try and see if it works.

Reply:
You're probably right. It's been a while since I did significant shell scripting.